convex/users.ts

```
@@ -135,6 +136,610 @@ export const getCurrentUserWithProfile = query({
+        updates.updatedAt = Date.now();
+
+        // Prefer internal patch that bypasses admin guard during controlled signup flow
+        await ctx.runMutation(internal.users.internalPatchProfile, {
+          profileId: (existing as any)._id,
+          updates,
+        } as any);
+        return { userId, profileId: (existing as any)._id };
+      }
+
+      // If we reached here, we already patched above. Return userId only.
+      return { userId };
```

Biome (2.1.2)

[error] This code will never be reached ...

... because this statement will return from the function beforehand


(lint/correctness/noUnreachable)